plugins/techdocs-backend/src/search/DefaultTechDocsCollator.test.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  PluginEndpointDiscovery,
+  getVoidLogger,
+} from '@backstage/backend-common';
+import { Entity } from '@backstage/catalog-model';
+import { DefaultTechDocsCollator } from './DefaultTechDocsCollator';
+
+const logger = getVoidLogger();
+
+const mockSearchDocIndex = {
+  config: {
+    lang: ['en'],
+    min_search_length: 3,
+    prebuild_index: false,
+    separator: '[\\s\\-]+',
+  },
+  docs: [
+    {
+      location: '',
+      text: 'docs docs docs',
+      title: 'Home',
+    },
+    {
+      location: 'local-development/',
+      text: 'Docs for first subtitle',
+      title: 'Local development',
+    },
+    {
+      location: 'local-development/#development',
+      text: 'Docs for sub-subtitle',
+      title: 'Development',
+    },
+  ],
+};
+
+const expectedEntities: Entity[] = [
+  {
+    apiVersion: 'backstage.io/v1alpha1',
+    kind: 'Component',
+    metadata: {
+      name: 'test-entity-with-docs',
+      description: 'Documented description',
+      annotations: {
+        'backstage.io/techdocs-ref': './',
+      },
+    },
+    spec: {
+      type: 'dog',
+      lifecycle: 'experimental',
+      owner: 'someone',
+    },
+  },
+  {
+    apiVersion: 'backstage.io/v1alpha1',
+    kind: 'Component',
+    metadata: {
+      name: 'test-entity',
+      description: 'The expected description',
+    },
+    spec: {
+      type: 'some-type',
+      lifecycle: 'experimental',
+      owner: 'someone',
+    },
+  },
+];
+
+jest.mock('cross-fetch', () => ({
```

Comment:
I think it's [recommended to do mocking of http requests using `msw`](https://backstage.io/docs/architecture-decisions/adrs-adr007)